### DISCUSSION TITLE
gost几个协议传输速度对比

### DISCUSSION THREAD

**Discussion by wpzzz** [go-gost/gost#496] [General]
本机iperf3测试，甲骨文amd机器，GOST v3.0.0-nightly.20240426
tcp转发不知为何失败，其他协议正常
结论，不考虑加密http最快，其次socks4和ss，加密的话就是纯tls，隧道用tls+relay。

ss aes-128-gcm
[  5]   0.00-10.00  sec   611 MBytes   513 Mbits/sec   27             sender
[  5]   0.00-10.07  sec   593 MBytes   495 Mbits/sec                  receiver

ss aes-128-cfb
[  5]   0.00-10.00  sec   369 MBytes   309 Mbits/sec   19             sender
[  5]   0.00-10.04  sec   360 MBytes   301 Mbits/sec                  receiver

ss chacha20-ietf-poly1305
[  5]   0.00-10.00  sec   449 MBytes   376 Mbits/sec   12             sender
[  5]   0.00-10.07  sec   436 MBytes   363 Mbits/sec                  receiver

ss chacha20
[  5]   0.00-10.00  sec   498 MBytes   417 Mbits/sec   18             sender
[  5]   0.00-10.07  sec   481 MBytes   401 Mbits/sec                  receiver

ss aes-128-ctr
[  5]   0.00-10.00  sec   392 MBytes   329 Mbits/sec   26             sender
[  5]   0.00-10.04  sec   383 MBytes   320 Mbits/sec                  receiver

http
[  5]   0.00-10.00  sec  1.66 GBytes  1.42 Gbits/sec   45             sender
[  5]   0.00-10.08  sec  1.64 GBytes  1.40 Gbits/sec                  receiver

socks
[  5]   0.00-10.00  sec   739 MBytes   620 Mbits/sec   23             sender
[  5]   0.00-10.02  sec   727 MBytes   609 Mbits/sec                  receiver

ss
[  5]   0.00-10.06  sec  1.35 GBytes  1.16 Gbits/sec  206             sender
[  5]   0.00-10.12  sec  1.35 GBytes  1.14 Gbits/sec                  receiver

http2
[  5]   0.00-10.00  sec   464 MBytes   389 Mbits/sec   23             sender
[  5]   0.00-10.08  sec   455 MBytes   379 Mbits/sec                  receiver

pht
[  5]   0.00-10.00  sec   110 MBytes  92.3 Mbits/sec   16             sender
[  5]   0.00-10.10  sec   101 MBytes  83.6 Mbits/sec                  receiver

tls
[  5]   0.00-10.00  sec   812 MBytes   682 Mbits/sec   28             sender
[  5]   0.00-10.09  sec   809 MBytes   672 Mbits/sec                  receiver

http+tls
[  5]   0.00-10.00  sec   779 MBytes   653 Mbits/sec   23             sender
[  5]   0.00-10.16  sec   776 MBytes   641 Mbits/sec                  receiver

quic
[  5]   0.00-10.00  sec   212 MBytes   178 Mbits/sec   15             sender
[  5]   0.00-10.06  sec   204 MBytes   170 Mbits/sec                  receiver

kcp
[  5]   0.00-10.00  sec  31.2 MBytes  26.2 Mbits/sec    2             sender
[  5]   0.00-10.11  sec  25.4 MBytes  21.1 Mbits/sec                  receiver

mtls
[  5]   0.00-10.00  sec   589 MBytes   494 Mbits/sec   29             sender
[  5]   0.00-10.05  sec   576 MBytes   481 Mbits/sec                  receiver

relay+tls
[  5]   0.00-10.00  sec   700 MBytes   587 Mbits/sec   26             sender
[  5]   0.00-10.11  sec   689 MBytes   572 Mbits/sec                  receiver

ws
[  5]   0.00-10.00  sec   454 MBytes   381 Mbits/sec   25             sender
[  5]   0.00-10.06  sec   436 MBytes   363 Mbits/sec                  receiver

relay+h2
[  5]   0.00-10.00  sec   428 MBytes   359 Mbits/sec   25             sender
[  5]   0.00-10.05  sec   419 MBytes   350 Mbits/sec                  receiver

h3
[  5]   0.00-10.07  sec  55.0 MBytes  45.8 Mbits/sec   27             sender
[  5]   0.00-10.11  sec  46.8 MBytes  38.9 Mbits/sec                  receiver

grpc
[  5]   0.00-10.01  sec   359 MBytes   301 Mbits/sec   15             sender
[  5]   0.00-10.01  sec   351 MBytes   294 Mbits/sec                  receiver

直连 不经过gost
[  5]   0.00-10.00  sec  3.43 GBytes  2.94 Gbits/sec  126             sender
[  5]   0.00-10.06  sec  3.43 GBytes  2.93 Gbits/sec                  receiver

relay
[  5]   0.00-10.01  sec  1.06 GBytes   914 Mbits/sec  107             sender
[  5]   0.00-10.09  sec  1.05 GBytes   892 Mbits/sec                  receiver

mtcp
[  5]   0.00-10.00  sec   424 MBytes   355 Mbits/sec   19             sender
[  5]   0.00-10.08  sec   415 MBytes   346 Mbits/sec                  receiver

 dtls
[  5]   0.00-10.06  sec  97.5 MBytes  81.3 Mbits/sec   18             sender
[  5]   0.00-10.12  sec  52.3 MBytes  43.3 Mbits/sec                  receiver

h2c
[  5]   0.00-10.00  sec   735 MBytes   616 Mbits/sec   34             sender
[  5]   0.00-10.10  sec   726 MBytes   603 Mbits/sec                  receiver

socks4
[  5]   0.00-10.00  sec  1.56 GBytes  1.34 Gbits/sec   51             sender
[  5]   0.00-10.10  sec  1.55 GBytes  1.32 Gbits/sec                  receiver

**Comment by cnwhy** [comment]
后期能优化下资源占用就好了，我家的软路由比较渣；
走 passwall 的 socks 正常，走 gost 的 socks CPU占用高很多，导致跑不满带宽

**Comment by ginuerzh** [comment]
对于SOCKS5，如果客户端用的也是GOST，默认会协商使用TLS加密，可以在客户端设置`notls=true`来关闭。
```
-F socks://:1080?notls=true
```

KCP的速度受其参数配置的影响很大，默认配置速度比较慢。

**Reply by wpzzz** [reply]
关闭之后确实快不少
[  5]   0.00-10.00  sec  1.16 GBytes   999 Mbits/sec  105             sender
[  5]   0.00-10.10  sec  1.15 GBytes   978 Mbits/sec                  receiver
